python/GafferUI/ColorChooser.py

```diff
@@ -620,6 +648,29 @@ def colorChangedSignal( self ) :
 
 		return self.__colorChangedSignal
 
+	## A signal emitted whenver the visible components are changed. Slots
+	# should have the signature slot( ColorChooser, visibleComponents ).
+	# `visibleComponents` is a string representing the components currently
+	# visible.
+	def visibleComponentsChangedSignal( self ) :
+
+		return self.__visibleComponentsChangedSignal
+
+	## A signal emitted whenver the static component is changed. Slots
+	# should have the signature slot( ColorChooser, staticComponent ).
+	# `staticComponent` is a single character string representing the
+	# current static component.
+	def staticComponentChangedSignal( self ) :
+
+		return self.__staticComponentChangedSignal
+
+	## A signal emitted whenever the visibility of the color field changes.
+	# Slots should have the signature slot( ColorChooser, visible ).
+	# `visible` is a boolean representing the current visibility.
```

Comment:
I wonder how useful the property (`staticComponent` etc) arguments to the slots are, given that the property can always be queried directly from the ColorChooser.

Reply:
Yeah, not much use in those extra arguments. Fixed in https://github.com/GafferHQ/gaffer/pull/6003/commits/7803fb8c31eb830efe67507b45d926e237015fb0